### PR TITLE
clang/tidy: Fixes (the rest)

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @wbpcode this gets the codebase (except mobile and a broken tool) passing clang-tidy

im not sure about some of the fixes so it will need to be checked carefully

anything i was particularly unsure about, i used a `NOLINT`

File: test/common/http/http1/codec_impl_test.cc [hunk 1]
```diff
@@ -4641,19 +4641,14 @@ TEST_P(Http1ServerConnectionImplTest, SeparatorInHeaderName) {
 // BalsaParser always rejects a header name with space. HttpParser only rejects
 // it in strict mode, which is disabled when ENVOY_ENABLE_UHV is defined.
 TEST_P(Http1ClientConnectionImplTest, SpaceInHeaderName) {
-  bool accept = parser_impl_ == Http1ParserImpl::HttpParser;
```
Comment (phlax) [inline test/common/http/http1/codec_impl_test.cc:4644]: check these

**Comment by phlax** [comment]
separating out PRs for source/contrib/test

**Comment by phlax** [comment]
@wbpcode this contains the remaining fixes/etc - a lot of `NOLINT` as its mostly issues i was less clear how to resolve

with the help of @ravenblackx ive managed to improve some of the fixes, but it would be good to remove more nolints - happy to iterate this before we land to get fixes right

**Review by wbpcode — COMMENTED** [review]
LGTM overall. only one quesion.

File: .clang-tidy [hunk 2]
```diff
@@ -67,11 +94,21 @@ CheckOptions:
...
 - key: readability-identifier-naming.TypeAliasCase
   value: 'CamelCase'
+- key: readability-identifier-naming.TypeAliasIgnoredRegexp
+  value: '(result_type)'
 - key: readability-identifier-naming.UnionCase
   value: 'CamelCase'
 - key: readability-identifier-naming.FunctionCase
   value: 'camelBack'
 
+HeaderFilterRegex: '^./source/.*|^./contrib/.*|^./test/.*'
```
[the remark this answers is not in the document]
Comment (phlax) [reply .clang-tidy]: reran and updated with fixes

File: envoy/stats/refcount_ptr.h [hunk 3]
```diff
@@ -115,7 +116,7 @@ template <class T> class RefcountPtr {
 // forth.
 #if __cplusplus < 202002L
 template <class T> static bool operator==(std::nullptr_t, const RefcountPtr<T>& a) {
-  return a == nullptr;
+  return a == nullptr; // NOLINT(clang-analyzer-cplusplus.Move)
```
Comment (wbpcode) [inline envoy/stats/refcount_ptr.h:119]: weird, why this line need this nolint...

**Review by wbpcode — DISMISSED** [review]
LGTM. Thanks. We add lots of NOLINT to the code base. But I think it is ok. The main responsibility of this PR is fix the CI rather than to improve the code quality. The last one could be achieved gradually by all contributors' work.

**Comment by wbpcode** [comment]
re to the coverage failure, seems the new `nolint` comments changed the coverage calculating result.

If you think it's OK, I change try to create a patch to your branch to update the code and avoid the `nolint` comments.
cc @phlax

**Comment by phlax** [comment]
> If you think it's OK, I could try to create a patch to your branch to update the code and avoid the nolint

that would be amazing

File: envoy/stats/refcount_ptr.h (hunk 3 above)
Comment (phlax) [reply envoy/stats/refcount_ptr.h:119]: no idea tbh - but confirmed its necessary

```console
./envoy/stats/refcount_ptr.h:119:10: error: Method called on moved-from object 'rp4' [clang-analyzer-cplusplus.Move,-warnings-as-errors]
  return a == nullptr;
         ^
```

https://dev.azure.com/cncf/envoy/_build/results?buildId=150324&view=logs&j=b7634614-24f3-5416-e791-4f3affaaed6c&t=d565ee52-b888-5d80-7fea-dfaed104c390&l=35

Comment (wbpcode) [reply envoy/stats/refcount_ptr.h:119]: also don't know why, orz.

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**